disco_aws_automation/disco_deploy.py

```diff
@@ -51,9 +52,15 @@ def _get_hostclasses_from_pipeline_definition(self, pipeline_definition):
         hostclasses = {entry["hostclass"]: entry for entry in pipeline_definition}
 
         for hostclass, entry in hostclasses.iteritems():
-            for definition in ["max_size", "desired_size", "min_size"]:
-                if definition in entry:
-                    hostclasses[hostclass][definition] = int(entry[definition])
+            if "min_size" in entry:
```

Comment:
@defyrlt : "desired_size" and "max_size" needs to get the max value from the set of available sizes, whereas "min_size" needs to get the min value from the available values.